chat-microservice/app/main.py

```diff
@@ -27,3 +29,16 @@ def create_application() -> FastAPI:
 
 
 app = create_application()
+
+@app.middleware("http")
+async def request_middleware(request: Request, call_next):
+    Logging.info(f"Request started at {request.url.path}")
+    response = await call_next(request)
+    response_body = [chunk async for chunk in response.body_iterator]
+    response.body_iterator = iterate_in_threadpool(iter(response_body))
+    if response.status_code >= 400:
+        Logging.error(f"{response.status_code} {json.loads(response_body[0].decode())['detail']} at {request.url.path}")
+    else:
+        Logging.info(f"{response.status_code} Request OK at {request.url.path}")
+    Logging.info(f"Request ended at {request.url.path}")
+    return response
```

Comment:
Could you refactor the middleware such that it resides in its own file in a package called “middlewares”?